pkg/controllerutils/miscellaneous.go

```diff
@@ -124,3 +128,25 @@ func ReconcileOncePer24hDuration(clock clock.Clock, objectMeta metav1.ObjectMeta
 
 	return 0
 }
+
+// GetMainReconciliationContext returns context with timeout for main client. Timeout should be equal to timeout passed in the argument but
+// not more than the DefaultReconciliationTimeout.
+func GetMainReconciliationContext(ctx context.Context, timeout time.Duration) (context.Context, context.CancelFunc) {
```

Comment:
Can you make sure to use the new helper functions consistently across the entire codebase? I.e., please use them in:
- pkg/operator/controller
- pkg/resourcemanager/controller

Reply:
Earlier `resourcemanager health` reconciler and `progressing` reconciler was using `2 minutes` for the main reconciler context timeout.
https://github.com/gardener/gardener/blob/3b33e4e25302bdfe254b9187b2cfe4c7885e43b9/pkg/resourcemanager/controller/health/health/reconciler.go#L63
https://github.com/gardener/gardener/blob/a97c75dcc483897b66af6ea71112208e9ded229c/pkg/resourcemanager/controller/health/progressing/reconciler.go#L55
With this change `health` and `progressing` reconciler will have `1 minute` as the main reconciler context timeout in consistent with the other reconcilers.